exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/okhttp/JsonRequestBody.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.internal.okhttp;
+
+import io.opentelemetry.exporter.internal.marshal.Marshaler;
+import io.opentelemetry.exporter.internal.marshal.MarshalerUtil;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import okhttp3.MediaType;
+import okhttp3.RequestBody;
+import okio.BufferedSink;
+
+public class JsonRequestBody extends RequestBody {
+  private static final MediaType JSON_MEDIA_TYPE = MediaType.parse("application/json");
+
+  private final byte[] preserializedJson;
+  private final int contentLength;
+
+  /** Creates a new {@link JsonRequestBody}. */
+  public JsonRequestBody(Marshaler marshaler) {
+    preserializedJson =
+        MarshalerUtil.preserializeJsonFields(marshaler).getBytes(StandardCharsets.UTF_8);
```

Comment:
Initially I wanted to save off the marshaler like ProtoResponseBody does and substitute `writeBinaryTo()` with `writeJsonTo()` in the overridden `writeTo()` method below. However there wasn't an equivalent method for getting the `contentLength` like ProtoResponseBody does in its constructor, so this seemed like the simplest way to do this with the existing code at my disposal.